python/cudf/cudf/_lib/column.pyx

```diff
@@ -203,9 +206,21 @@ cdef class Column:
             "The value for mask is smaller than expected, got {}  bytes, "
             "expected " + str(required_num_bytes) + " bytes."
         )
+
+        # Because hasattr will trigger invocation of
+        # `__cuda_array_interface__` which could
+        # be expensive in CopyOnWriteBuffer case.
+        value_cai = inspect.getattr_static(
+            value,
+            "__cuda_array_interface__",
+            None
+        )
```

Comment:
What about something like this?
```suggestion
        has_cai = isinstance(value, Buffer) or hasattr(value, "__cuda_array_interface__")
```

Or even better, we could create a small PR that implement an util function to access `__cuda_array_interface__`, which we could use exclusively. This would also simplify `SpillableBuffer`.

Reply:
I'm planning to add a utility to remove this code duplication across the repo.